mono/eglib/glib.h

```diff
@@ -54,6 +54,248 @@
 #define G_END_DECLS
 #endif
 
+#ifdef __cplusplus
+
+#define g_cast_t monoeg_g_cast_t // in case not inlined (see eglib-remap.h)
+
+// Mostly-internal type to support g_cast(), but also can be used in containers.
+template <typename TFrom> // TFrom is almost always void*
+struct g_cast_t
+{
+private:
+	TFrom x;
+public:
+	explicit g_cast_t (TFrom y) : x(y)
+	{
+	}
+
+	g_cast_t () // Not really intended but has a user.
```

Comment:
If I understand what this is for (to add `g_cast_t<>` data members to previously C-compatible structs), I think we're better off without it.

Reply:
It avoids casts in uses of lists. For actual line damage reduction not to provide a well designed type. Same thing with most of the operators, including both orders. They might not all be used, but quite a few are. Including ordering, not just equality.